language/stdlib/modules/DiemAccount.move

```diff
@@ -1641,22 +1637,22 @@ module DiemAccount {
 
         // Don't grab the balance if the transaction fee is zero
         if (max_transaction_fee > 0) {
-            // [PCA6]: Check that the account has a balance in this currency
+            // [PCA6]: Check that the gas fee can be paid in this currency
```

Comment:
Another question: could this check be performed even when the max fee is zero? It would be simpler to specify that the gas currency must always be registered for transaction fees, instead of having to say that the check is conditional upon a non-zero max fee. I left it here for 2 reasons: (1) it would be a breaking change for a txn with an invalid gas currency and a gas price of zero, and (2) this gives an automatic exception to WriteSet transactions, which could be important if the TransactionFee currencies ever get broken and need to be fixed. It's not terrible to leave this here, but I welcome suggestions for anything better.

Reply:
The check could be performed when the max fee is zero. 

However, I would be in favor of not doing that for the reasons you mention. I would also add two additional reasons for this:
1. Not checking that the currency is a gas-fee currency seems most in-line (at least to me) with the logic that we have around gas in the "fee = 0" case in the prologue. In that case we don't check that the sending account has a balance in the currency. 
2. If we view the prologue as the minimal set of preconditions for successful execution of the epilogue (with the exception of someone lowering their balance to below the transaction fee during the transaction), not checking the gas currency is a gas-fee currency also makes sense with the logic in https://github.com/diem/diem/blob/master/language/stdlib/modules/DiemAccount.move#L1783